be/src/storage/rowset/binary_plain_page.cpp

```diff
@@ -199,6 +193,171 @@ void BinaryPlainPageDecoder<Type>::batch_string_at_index(Slice* dst, const int32
 #endif
 }
 
+template <LogicalType Type>
+Status BinaryPlainPageDecoder<Type>::next_batch_with_filter(
+        Column* column, const SparseRange<>& range, const std::vector<const ColumnPredicate*>& compound_and_predicates,
+        const uint8_t* null_data, uint8_t* selection, uint16_t* selected_idx) {
+    DCHECK(null_data == nullptr || column->is_nullable());
+    DCHECK(_parsed);
+    if (PREDICT_FALSE(_cur_idx >= _num_elems)) {
+        return Status::OK();
+    }
+
+    size_t to_read = std::min(range.span_size(), _num_elems - _cur_idx);
+    SparseRangeIterator<> iter = range.new_iterator();
+    bool append_status = true;
+    size_t index = 0;
+    while (to_read > 0) {
+        _cur_idx = iter.begin();
+        Range<> r = iter.next(to_read);
+        size_t length = r.span_size();
+        size_t end = _cur_idx + length;
+        append_status &= next_range_with_filter(_cur_idx, end, column, compound_and_predicates,
+                                                null_data != nullptr ? null_data + index : nullptr, selection + index,
+                                                selected_idx + index);
+        index += length;
+        to_read -= length;
+        _cur_idx = end;
+    }
+    if (append_status) {
+        return Status::OK();
+    }
+
+    return Status::InternalError("BinaryPlainPageDecoder::next_batch_with_filter error!");
+}
+
+template <LogicalType Type>
+bool BinaryPlainPageDecoder<Type>::next_range_with_filter(
```

Comment:
suggest to use Status instead of bool as return type, since the return value means success or failure.